File: agent/evaluation/EvalCase.py
from __future__ import annotations

from dataclasses import dataclass, field

from json_types import JsonObject


@dataclass
class EvalCase:
    """单个评估用例。"""

    id: str
    query: str
    expected_chunks: list[str] = field(default_factory=list)
    expected_answer: str = ""
    expected_annotation: JsonObject = field(default_factory=dict)
    tags: list[str] = field(default_factory=list)

File: agent-ts/src/evaluation/model/EvalCase.ts
import type { JsonObject } from "../../common/json/types/JsonTypes.js";

export interface EvalCase {
  readonly id: string;
  readonly query: string;
  readonly expectedChunks: readonly string[];
  readonly tags: readonly string[];
  readonly expectedAnnotation?: JsonObject | null;
  readonly expectedAnswer?: string | null;
}
